### PR TITLE
fix(document): use <section>s with aria-labelledby

### PR DESCRIPTION
**Pull request by caugner**
## Summary

Makes sure we use `<section>` elements with ~~`aria-describedby`~~ `aria-labelledby` in MDN articles.

### Problem

Previously, we were just using `<div>`s to wrap MDN article sections, which is not ideal.

### Solution

I replaced the `<div>`s with `<section>`s and added ~~`aria-describedby`~~ `aria-labelledby` where possible.

---

## Screenshots

_No visual changes._

---

## How did you test this change?

I did not see any differences, **but** I did not check for potential CSS rules targeting those `div`s.

### CODE REVIEW

File: client/src/document/ingredients/prose.tsx [hunk 1]
```diff
@@ -1,10 +1,21 @@
...
+  return (
+    <section
+      aria-describedby={id}
+      dangerouslySetInnerHTML={{ __html: section.content }}
+    />
+  );
 }
 
-export function ProseWithHeading({ id, section }) {
+export function ProseWithHeading({
```
Comment (caugner) [inline client/src/document/ingredients/prose.tsx:12]: I wonder whether we couldn't just merge `Prose` and `ProseWithHeading` now that both accept the id parameter. 🤔

Comment (schalkneethling) [reply client/src/document/ingredients/prose.tsx:12]: I would be open to that.

**Comment by schalkneethling** [comment]
> Talked about this with @fiji-flo last week, and what we probably want is to have the sections contain the heading. Then we would only have sections within an article, and the section would contain the heading next to the other items (paragraphs, tables).

What I was really getting at is, we might not need `section` at all. Without using either `aria-label` or `aria-labelledby` a `section` element is pretty much the same as using a `div`. See this article for reference: https://www.smashingmagazine.com/2020/01/html5-article-section/#so-what-about-lt-section-gt

**Comment by schalkneethling** [comment]
> > Talked about this with @fiji-flo last week, and what we probably want is to have the sections contain the heading. Then we would only have sections within an article, and the section would contain the heading next to the other items (paragraphs, tables).
> 
> What I was really getting at is that we might not need `section` at all. Without using either `aria-label` or `aria-labelledby` a `section` element is pretty much the same as using a `div`. See this article for reference: https://www.smashingmagazine.com/2020/01/html5-article-section/#so-what-about-lt-section-gt

Also, creating too many `regions` or landmarks on a page can actually be less helpful to screen reader users.

**Comment by caugner** [comment]
> Without using either `aria-label` or `aria-labelledby` a `section` element is pretty much the same as using a `div`.

Oh, good catch! I added `aria-describedby` in this PR for what should have been `aria-labelled-by`.

@schalkneethling Wouldn't this improve the situation, having labelled sections exclusively inside the article?

**Comment by schalkneethling** [comment]
> @schalkneethling Wouldn't this improve the situation, having labelled sections exclusively inside the article?

It might but as mentioned, some of these sections simply wrap a single paragraph sometimes. We would have to see how common that is. Just want to make sure we do not make it too noisy by adding all these regions to the page.

**Comment by schalkneethling** [comment]
@caugner What would be the resulting DOM if we do not wrap these with anything? i.e. not a `div` nor a `section` element. Perhaps there are style hooks that rely on this though?

**Comment by caugner** [comment]
@schalkneethling I have now ...

1. Merged `Prose` with `ProseWithHeading`.
2. Replaced `aria-describedby` with `aria-labelledby`.
3. Moved the heading inside the section.

The resulting `<article>` structure is as follows:

image

It's not perfect yet, because our document structure is not hierarchical (e.g. the two Example regions belong to the "Accessibility concerns", but are not inside that section), but I think this is an improvement over the status quo, where the content of each section is not associated with the corresponding heading.

File: client/src/document/ingredients/prose.tsx (hunk 1 above)
Comment (caugner) [reply client/src/document/ingredients/prose.tsx]: Done in 1ba27e1.

**Review by schalkneethling — APPROVED** [review]
Thanks so much for this, @caugner 🎉

**Pull request merged**